src/main/scala/org/broadinstitute/dsde/workbench/sam/util/SamRequestContext.scala

```diff
@@ -0,0 +1,10 @@
+package org.broadinstitute.dsde.workbench.sam.util
+
+import io.opencensus.trace.Span
+
+/*
+SamRequestContext was created with the goal of pre-empting a repeat of the crazy plumbing that was required to add
+the trace. In the future, instead of needing to add a new param to each API and underlying function, we can just add
+to this class instead.
+ */
+case class SamRequestContext (parentSpan: Option[Span])
```

Comment:
I don't know how Scala case classes work, but could we have a factory function for creating the SamRequestContext from a Span and one for from None? I think passing None is not really a "normal" use, and having a factory function to do it can represent that.

SamRequestContext contextForTest() {...}

Reply:
I think there are certainly places where we don't want tracing. The status api call comes to mind because they will only generate noise (something like 80% of our api calls are to the status api which means we would have to turn up our sampling rate in order to get a reasonable sample of the other api calls.

Reply:
I also thought it would be nice to have a factory function for contexts with no traces. I have a shelved branch toward that goal. It was turning into a more-than-trivial amount of work and doug recommended that we use Option[Span] instead. I can reopen that if we want to go that route?

Reply:
Option[Span] is ok, but let's comment on why we might not have a Span.
If there are multiple reasons, a factory function is harder to name. What might get interesting is when/if we add a second variable to SamRequestContext. Then we might want to restrict how it's built. But if it's more-than-trivial, we can wait until then and eat the work later.

If we're sampling with a probability, whether we're sampling the status API doesn't affect any other API. All calls have the same likelihood of being sampled; they're all independent. Sampling status will just cause us to have more traces, so just cost us more to store. So if we don't want to do that, sounds ok.